operator/builtin/input/http/operator.go

```diff
@@ -0,0 +1,216 @@
+package httpevents
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/gorilla/mux"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/observiq/stanza/entry"
+	"github.com/observiq/stanza/operator"
+	"github.com/observiq/stanza/operator/helper"
+)
+
+func init() {
+	operator.Register("http_input", func() operator.Builder { return NewHTTPInputConfig("") })
+}
+
+// HTTPInput is an operator that listens for log entries over http.
+type HTTPInput struct {
+	helper.InputOperator
+
+	server      http.Server
+	json        jsoniter.API
+	maxBodySize int64
+
+	auth authMiddleware
+
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
+}
+
+// Start will start listening for log entries over http.
+func (t *HTTPInput) Start() error {
+	ctx, cancel := context.WithCancel(context.Background())
+	t.cancel = cancel
+	t.goListen(ctx)
+	return nil
+}
+
+// Stop will stop listening for log entries over http.
+func (t *HTTPInput) Stop() error {
+	t.cancel()
+	t.wg.Wait()
+	return nil
+}
+
+// goListenn will listen for http connections.
+func (t *HTTPInput) goListen(ctx context.Context) {
+	t.Debugf("using server config: %d", t.server.MaxHeaderBytes)
+
+	t.wg.Add(1)
+
+	entryCreateMethods := []string{"POST", "PUT"}
+
+	m := mux.NewRouter()
+	m.HandleFunc("/", t.goHandleMessages).Methods(entryCreateMethods...)
+
+	if t.auth != nil {
+		t.Debugf("using authentication middleware: %s", t.auth.name())
+		m.Use(t.auth.auth)
+	}
+
+	m.HandleFunc("/health", t.health).Methods("GET")
+
+	t.server.Handler = m
+
+	// shutdown go routine waits for a canceled context before stopping the server
+	go func() {
+		defer t.wg.Done()
+		for {
```

Comment:
You don't need a for loop with a default statement to accomplish this. You can remove all of this and replace it with just `<-ctx.Done()` followed by the shutdown code. This will block until the context is cancelled and then it will execute the code that comes after it.